renderer/src/components/FilAddressForm.tsx

```diff
@@ -1,57 +1,85 @@
 import { FormEvent, useEffect, useState } from 'react'
 import { checkAddressString } from '@glif/filecoin-address'
 
+interface ValidationErrorProps {
+  message: string | undefined
+}
+
+const ValidationError: FC<ValidationErrorProps> = ({ message }) => {
+  return (
+    message
+      ? <span className="err-msg" title="validation error">{message}</span>
+      : <span className='mb-[0.625rem]'>&nbsp;</span>
+  )
+}
+
 interface FilAddressFormProps {
-    address?: string
-    setAddress: React.Dispatch<React.SetStateAction<string | undefined>>
-    onValidAddress: (address: string) => void
+  setFilAddress: (address: string | undefined) => void
 }
 
-export default function FilAddressForm (props: FilAddressFormProps) {
-  const backendAddress = props.address ?? ''
-  const [address, setAddress] = useState(backendAddress)
+const FilAddressForm: FC<FilAddressFormProps> = ({ setFilAddress }) => {
   const [validationError, setValidationError] = useState<string | undefined>()
+  const [inputAddr, setInputAddr] = useState<string>('')
 
   const validateAddress = () => {
     setValidationError(undefined)
-    if (!address) return true
     try {
-      checkAddressString(address)
+      checkAddressString(inputAddr)
       return true
     } catch (err) {
       setValidationError(err instanceof Error ? err.message : '' + err)
       return false
     }
   }
 
-  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
+  const handleChangeAddress = (event: ChangeEvent<HTMLInputElement>) => {
+    setInputAddr(event.target.value)
+  }
 
-    if (validateAddress() && address !== props.address) {
-      props.onValidAddress(address)
+  const handleAuthenticate = (event: React.SyntheticEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    if (!inputAddr) {
+      return true
+    }
+    if (validateAddress()) {
+      setFilAddress(inputAddr)
```

Comment:
```suggestion
    if (inputAddr && validateAddress()) {
      setFilAddress(inputAddr)
```

Iiuc `return true` doesn't do anything here, since we already called `event.preventDefault()`